backend/src/main/java/com/example/dontwastetomuch/game/GameController.java

```diff
@@ -158,4 +157,23 @@ public ResponseEntity<CommunityStatsDTO> getOneCommunityGame(@PathVariable Strin
             return new ResponseEntity(e.getMessage(), HttpStatus.NOT_FOUND);
         }
     }
+
+    @GetMapping("/edit/{gameId}")
+    public ResponseEntity<Game> getOneGameToEdit(@PathVariable String gameId, Principal principal){
+        try{
+            return ResponseEntity.ok(gameService.getOneGameToEdit(gameId, principal));
```

Comment:
Same here. No need to send the principal, just the username from principal.getName().
getOneGameToEdit can also throw IllegalArgumentExceptions which are not caught by the controller